runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -950,8 +1002,10 @@ public DataflowPipelineJob run(Pipeline pipeline) {
             .addAllCapabilities(Environments.getJavaCapabilities())
             .build());
 
-    RunnerApi.Pipeline portablePipelineProto =
+    RunnerApi.Pipeline pipelineProto =
         PipelineTranslation.toProto(pipeline, portableComponents, false);
+    List<DataflowPackage> packages = stageArtifacts(pipelineProto);
```

Comment:
Add a comment that this has to be invoked before the "resolveArtifacts" call.

Reply:
Done.